### PR TITLE
#182201676 ft-signup: allow users to create accounts in the system

### PR DESCRIPTION
**Pull request by uwambajeddy**
##  What does this PR do?
- Allow user to create new accounts
##  Description of Task to be completed?
- fetch `API` from the backend
## How should this be manually tested?
- Clone the repo.
- Run `npm install` to install modules.
- Run `npm start` to run an app.
- Navigate to `/signup` and create an account

## What are the relevant pivotal tracker stories?
 - https://www.pivotaltracker.com/n/projects/2569105/stories/182201676

### CODE REVIEW

**Review by Placide123 — CHANGES_REQUESTED** [review]
on my end it is not running other can check

**Comment by uwambajeddy** [comment]
> on my end it is not running other can check

Please let's be professional if you  `request for changes` it means you also indicate the issue, how do you think I am supposed to help you?

File: src/views/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                        <div className="form-group">
+                            <div className="input-group input-resize" id='username'>
+                                <input type="text" name="" placeholder="Username" value={userRequest.username} onChange={(event)=> setUserRequest({...userRequest,userName: event.target.value})}  />
+                                <ion-icon name="person"></ion-icon>
+                            </div>
+                            <small id="username-error" className="error"></small>
+                        </div>
+                        <div className="form-group input-left">
+                            <div className="input-group input-resize  " id='email'>
+                                <input type="email" name="" placeholder="Email" value={userRequest.email} onChange={(event)=> setUserRequest({...userRequest,email: event.target.value})}  />
```
Comment (joshwambere) [inline src/views/Signup.jsx]: why did you not use stories for input while you created a story component

File: src/views/Signup.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                                <option data-testid="select-option" value="female">Female</option>
+                              </select>
+                            </div>
+                        <small id="gender-error" className="error"></small>
+                        </div>
+                        </div>
+                        <div className="form-group input-align">
+                            <div className={userRequest.buttonDisable? "input-group input-resize button button-disable": "input-group input-resize button"}>
+                                <input type="submit" value="Create account"/>
+                            </div>
```
Comment (joshwambere) [inline src/views/Signup.jsx]: also create a story for buttons

File: src/views/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+const Signup = () =>{
+  const navigate = useNavigate()
+  const [userRequest, setUserRequest] = useState({buttonDisable:false,email:'',lastName:'',role:'requester',phoneNumber:'',gender:'select',password:'',repeatPassword:'',firstName:'',userName:''});
+
+  const submitHandle = (e) => {
+     e.preventDefault();
+     if (signupValidation(userRequest.email, userRequest.password,userRequest.repeatPassword,userRequest.phoneNumber,userRequest.userName,userRequest.firstName,userRequest.lastName,userRequest.gender)) {
+        setUserRequest({...userRequest, buttonDisable:true})
+      axios.post(`${process.env.API_URL}/user/auth/signup`, {
```
Comment (joshwambere) [inline src/views/Signup.jsx:154]: use redux for handling request with the backend

File: src/views/Signup.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import barefootLogo from "../assets/barefoot_logo.svg";
+import { Link, useNavigate } from "react-router-dom";
+import React, { useState } from 'react';
+import axios from 'axios';
+import swal from 'sweetalert2';
+import signupValidation from '../service/validations/signup';
+
+const Signup = () =>{
+  const navigate = useNavigate()
+  const [userRequest, setUserRequest] = useState({buttonDisable:false,email:'',lastName:'',role:'requester',phoneNumber:'',gender:'select',password:'',repeatPassword:'',firstName:'',userName:''});
```
Comment (joshwambere) [inline src/views/Signup.jsx]: this kind of state will be a headache to maintain, group information which are related

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by semy-tuyi — APPROVED** [review]
Looks Good

**Review by Placide123 — CHANGES_REQUESTED** [review]
work on responsiveness 
Screen Shot 2022-08-18 at 11 52 36

and send is not working at my end
Screen Shot 2022-08-18 at 11 54 00

**Review by alaintresor — APPROVED** [review]

**Comment by Patricknshimiyimana** [comment]
> and send is not working at my end Screen Shot 2022-08-18 at 11 54 00

I think the problem is in your webpack configuration ask @uwambajeddy to give you his version of webpack  file he is using

File: src/service/validations/signup.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
+
```
Comment (joshwambere) [inline src/service/validations/signup.js]: those validations are a too much of vanilla javascript maybe take a look in this blog https://dev.to/codebucks/form-validation-in-reactjs-by-building-reusable-custom-hook-1bg7, or consider a third party form validation library

**Review by joshwambere — CHANGES_REQUESTED** [review]

**Review by semy-tuyi — APPROVED** [review]

**Review by Patricknshimiyimana — COMMENTED** [review]
The create account button is doing nothing on my side
It is not even submitting the form

**Review by alaintresor — APPROVED** [review]

**Review by Patricknshimiyimana — APPROVED** [review]

**Pull request merged**